src/gas/gas-util.test.ts

```diff
@@ -155,72 +227,741 @@ describe('gas utils', () => {
     });
   });
 
-  describe('normalizeGWEIDecimalNumbers', () => {
-    it('should convert a whole number to WEI', () => {
-      expect(normalizeGWEIDecimalNumbers(1)).toBe('1');
-      expect(normalizeGWEIDecimalNumbers(123)).toBe('123');
-      expect(normalizeGWEIDecimalNumbers(101)).toBe('101');
-      expect(normalizeGWEIDecimalNumbers(1234)).toBe('1234');
-      expect(normalizeGWEIDecimalNumbers(1000)).toBe('1000');
-    });
+  describe('fetchEthGasPriceEstimate', () => {
```

Comment:
This should include tests for the error case as well.

Reply:
What error case were you thinking of? 

For numbers, were you thinking what if the `gasPrice` is some big number? If so, `weiHexToGweiDec` should be able to handle big numbers correctly, so can we trust that it's doing the right thing?

Reply:
The `await query` call can throw, e.g. due to a network error.

>For numbers, were you thinking what if the `gasPrice` is some big number? If so, `weiHexToGweiDec` should be able to handle big numbers correctly, so can we trust that it's doing the right thing? If we are lacking tests around `weiHexToGweiDec`, then that's understandable, but maybe we double-check number support in those tests?

Right, but the unit tests should establish what the function is meant to do, irrespective of the current implementation. If/when we decide to replace `BN` for example, we'll want to ensure we don't unknowingly introduce a change here.

Reply:
Okay, I can add some overlapping test coverage.

I understand that the thing we know the least about, and can hence trust the least, is `ethQuery.gasPrice()` / `eth_gasPrice`. Do you think it is worth testing what happens if `gasPrice` is null, undefined, empty string, something that is not a hex string, etc.? I'd like to think we are able to make some assumptions here.

Reply:
Hmm. Good question. According to [the wiki](https://eth.wiki/json-rpc/API),  it should always return a "QUANTITY", i.e. a string hex number. So those should all be impossible. But better if we can avoid trusting third party APIs, especially when the user can set them to a custom node that doesn't follow these rules.

This is a good example of a case where extracting an API interaction into a separate module might be simpler. So that in cases like this, we can trust that the return type is what we expect. Then in our API module we have validation and throw if anything is wrong.

So... the point being, you raise a great point and we should handle that but perhaps it can be done elsewhere, in a dedicated API module or built in to the `query` function we're calling EthQuery with here, something like that.